### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'isLazyTypeFunc')

### ISSUE THREAD

**Issue by MihaiVoinea** [risen228/nestjs-zod#14]
Getting `TypeError: Cannot read properties of undefined (reading 'isLazyTypeFunc')` when I use `patchNestJsSwagger` and I try to apply a `ApiBadRequestResponse` decorator from @nestjs/swagger. Is there any way to extend the OpenAPI object through @nestjs/swagger decorators when using patchNestJsSwagger?

**Comment by therosbif** [comment]
@MihaiVoinea did you find a solution to this problem?

**Comment by MihaiVoinea** [comment]
@therosbif Nope, switched to another package.

**Comment by risenforces** [comment]
Will be fixed in #18

**Comment by HiroyukiNIshimura** [comment]
!isZodDto(type) calls defaultExplore, but exploreModelSchema is an instance function that internally refers to undefined 'this' and causes an error.

Since the function is extended with 'prototype', this.exploreModelSchema causes a stack over due to recursion 😅

It would be best if nestjs-swagger could pass a callback to exploreModelSchema, but this is how it worked for now!

```ts
import { Type } from '@nestjs/common';
import { SchemaObjectFactory } from '@nestjs/swagger/dist/services/schema-object-factory';
import { isZodDto } from 'nestjs-zod/dto';
import { zodToOpenAPI } from 'nestjs-zod';
import * as _ from 'lodash';

function getSchemaObjectFactory(): Type<SchemaObjectFactory> {
  return SchemaObjectFactory;
}

function isLazyTypeFunc(type) {
  return _.isFunction(type) && type.name == 'type';
}

export function patchNestJsSwagger(
  SchemaObjectFactory = getSchemaObjectFactory(),
) {
  if (SchemaObjectFactory.prototype.__patchedWithLoveByNestjsZod) return;

  SchemaObjectFactory.prototype.exploreModelSchema = function (
    this: SchemaObjectFactory,
    type,
    schemas,
    schemaRefsStack,
  ) {
    if (isLazyTypeFunc(type)) {
      const factory = type as () => Type<unknown>;
      type = factory();
    }

    if (!isZodDto(type)) {
      if (isLazyTypeFunc(type)) {
        type = type();
      }
      const propertiesWithType = this.extractPropertiesFromType(
        type,
        schemas,
        schemaRefsStack,
      );
      if (!propertiesWithType) {
        return '';
      }
      const typeDefinition = {
        type: 'object',
        properties: _.mapValues(
          _.keyBy(propertiesWithType, 'name'),
          (property) =>
            _.omit(property, ['name', 'isArray', 'required', 'enumName']),
        ),
      };
      const typeDefinitionRequiredFields = propertiesWithType
        .filter((property) => property.required != false)
        .map((property) => property.name);
      if (typeDefinitionRequiredFields.length > 0) {
        typeDefinition['required'] = typeDefinitionRequiredFields;
      }
      schemas[type.name] = typeDefinition;
      return type.name;
    }

    schemas[type.name] = zodToOpenAPI(type.schema);

    return type.name;
  };

  SchemaObjectFactory.prototype.__patchedWithLoveByNestjsZod = true;
}
```

**Comment by risenforces** [comment]
May be fixed in #42
